patches/tModLoader/Terraria/GameContent/UI/ResourceSets/ClassicPlayerResourcesDisplaySet.cs.patch

```diff
@@ -0,0 +1,231 @@
+ 
++			UIDisplay_LifePerHeart = snapshot.LifePerSegment;
++			int num2 = snapshot.LifeFruitCount;
++
++			bool drawHearts = ResourceOverlayLoader.PreDrawResourceDisplay(snapshot, this, true, ref color, out bool drawText);
++
++			if (!drawText)
++				goto SkipLifeText;
++
++			/*
+ 			int num = localPlayer.statLifeMax / 20;
+ 			int num2 = (localPlayer.statLifeMax - 400) / 5;
+ 			if (num2 < 0)
+@@ -52,33 +_,38 @@
+ 
+ 			int num3 = localPlayer.statLifeMax2 - localPlayer.statLifeMax;
+ 			UIDisplay_LifePerHeart += num3 / num;
++			*/
+-			int num4 = (int)((float)localPlayer.statLifeMax2 / UIDisplay_LifePerHeart);
++			int num4 = (int)((float)snapshot.LifeMax / UIDisplay_LifePerHeart);
+ 			if (num4 >= 10)
+ 				num4 = 10;
+ 
+-			string text = Lang.inter[0].Value + " " + localPlayer.statLifeMax2 + "/" + localPlayer.statLifeMax2;
++			string text = Lang.inter[0].Value + " " + snapshot.LifeMax + "/" + snapshot.LifeMax;
+ 			Vector2 vector = FontAssets.MouseText.Value.MeasureString(text);
+ 			if (!localPlayer.ghost) {
+ 				spriteBatch.DrawString(FontAssets.MouseText.Value, Lang.inter[0].Value, new Vector2((float)(500 + 13 * num4) - vector.X * 0.5f + (float)UI_ScreenAnchorX, 6f), color, 0f, default(Vector2), 1f, SpriteEffects.None, 0f);
+-				spriteBatch.DrawString(FontAssets.MouseText.Value, localPlayer.statLife + "/" + localPlayer.statLifeMax2, new Vector2((float)(500 + 13 * num4) + vector.X * 0.5f + (float)UI_ScreenAnchorX, 6f), color, 0f, new Vector2(FontAssets.MouseText.Value.MeasureString(localPlayer.statLife + "/" + localPlayer.statLifeMax2).X, 0f), 1f, SpriteEffects.None, 0f);
++				spriteBatch.DrawString(FontAssets.MouseText.Value, snapshot.Life + "/" + snapshot.LifeMax, new Vector2((float)(500 + 13 * num4) + vector.X * 0.5f + (float)UI_ScreenAnchorX, 6f), color, 0f, new Vector2(FontAssets.MouseText.Value.MeasureString(snapshot.Life + "/" + snapshot.LifeMax).X, 0f), 1f, SpriteEffects.None, 0f);
+ 			}
+ 
+-			for (int i = 1; i < (int)((float)localPlayer.statLifeMax2 / UIDisplay_LifePerHeart) + 1; i++) {
++			SkipLifeText:
++			if (!drawHearts)
++				goto SkipHearts;
++
++			for (int i = 1; i < snapshot.AmountOfLifeHearts + 1; i++) {
+ 				int num5 = 255;
+ 				float num6 = 1f;
+ 				bool flag = false;
+-				if ((float)localPlayer.statLife >= (float)i * UIDisplay_LifePerHeart) {
++				if ((float)snapshot.Life >= (float)i * UIDisplay_LifePerHeart) {
+ 					num5 = 255;
+-					if ((float)localPlayer.statLife == (float)i * UIDisplay_LifePerHeart)
++					if ((float)snapshot.Life == (float)i * UIDisplay_LifePerHeart)
+ 						flag = true;
+ 				}
+ 				else {
+-					float num7 = ((float)localPlayer.statLife - (float)(i - 1) * UIDisplay_LifePerHeart) / UIDisplay_LifePerHeart;
++					double num7 = ((float)snapshot.Life - (float)(i - 1) * UIDisplay_LifePerHeart) / UIDisplay_LifePerHeart;
+ 					num5 = (int)(30f + 225f * num7);
+ 					if (num5 < 30)
+ 						num5 = 30;
+ 
+-					num6 = num7 / 4f + 0.75f;
++					num6 = (float)(num7 / 4f + 0.75f);
```

Comment:
`num6` appears to just be used for filling scale due to the "pulse" effect, so it's probably fine to just have `num7` be a `float` variable instead of `double`
same idea for the mana drawing method